core/validation/casting.py

```diff
@@ -18,24 +20,15 @@ def cast_to_schema(data: dict[str, pd.DataFrame], schema: dict) -> None:
             continue  # skip casting if schema doesn't exist for that table_data - this will be caught during validation
 
         column_to_type = schema[table]["columns"]
-        table_types = table_data.dtypes
-
-        table_retyped = False
-        for column, target_type in column_to_type.items():
-            if column in table_types:
-                original_type = table_types[column]
-
-                if original_type != target_type:
-                    if target_type == "list":
-                        # Forcible conversion of values to target_type == "list" occurs in extract_postcodes()
-                        # and .astype does not support lists
-                        continue
-
-                    try:
-                        table_data[column] = table_data[column].astype(target_type)
-                        table_retyped = True
-                    except ValueError:
-                        continue  # this will be caught during validation
-
-        if table_retyped:
-            data[table] = table_data
+
+        for pos, (index, row) in enumerate(table_data.iterrows()):
+            for column, value in row.items():
+                # Postcodes is now a list
+                if pd.isna(value) or isinstance(value, (datetime, pd.Timestamp, list)):
```

Comment:
Do we not want to validate that a cell is a datetime, or am I missing something?